core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -203,6 +203,28 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     getOption(key).map(_.toBoolean).getOrElse(defaultValue)
   }
 
+  // Limit of bytes for total size of results (default is 1GB)
```

Comment:
Option 3. could be: Put such methods in a `SparkConfUtils` object that would be less prone to kitchen-sink-iness.

I'm impartial, I'll let you make the call between these three.